app/geo/gui/src/ts/twxplore/gui/geo/reducerFunctions/updateAttributeStatesOfFeatureType.ts

```diff
@@ -0,0 +1,50 @@
+import {MapFeatureAttributeState} from "../states/map/MapFeatureAttributeState";
+import {MapFeature} from "../states/map/MapFeature";
+import {getFeatureAttributeByName} from "../attributeStrategies/getFeatureAttributeByName";
+
+export const updateAttributeStatesOfFeatureType = (
+  attributeStatesOfFeatureType: {
+    [featureAttributeName: string]: MapFeatureAttributeState;
+  },
+  addedFeature: MapFeature
+) => {
+  //Loop through the feature's attribute
+  for (const attribute of Object.keys(addedFeature)) {
+    //If the arttribute is numeric
+    if (getFeatureAttributeByName(attribute).isNumeric) {
+      //If this is the first time coming across this attribute for this feature type
+      if (
+        !attributeStatesOfFeatureType[attribute].min &&
+        !attributeStatesOfFeatureType[attribute].min
+      ) {
+        //New max of the attribute wil be the feature's value for the attribute
+        attributeStatesOfFeatureType[attribute].max = (addedFeature as any)[
+          attribute
+        ];
+        //New min of the attribute wil be the feature's value for the attribute
+        attributeStatesOfFeatureType[attribute].min = (addedFeature as any)[
```

Comment:
Not sure about this addedFeature as any business.

I'd rather see something like:

const attributeKey = attributeName as keyof MapFeatureType.

(https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208)

Then you can dispense with this as any business.